api/v1alpha1/opentelemetrycollector_webhook.go

```diff
@@ -91,5 +93,18 @@ func (r *OpenTelemetryCollector) validateCRDSpec() error {
 		return fmt.Errorf("the OpenTelemetry Collector mode is set to %s, which does not support the attribute 'tolerations'", r.Spec.Mode)
 	}
 
+	// validate target allocation
+	if r.Spec.TargetAllocator.Enabled && r.Spec.Mode != ModeStatefulSet {
+		return fmt.Errorf("the OpenTelemetry Collector mode is set to %s, which does not support the target allocation deployment", r.Spec.Mode)
+	}
+
+	// validate Prometheus config for target allocation
+	if r.Spec.TargetAllocator.Enabled {
+		_, err := ta.ConfigToPromConfig(r.Spec.Config)
+		if err != nil {
+			return fmt.Errorf("the OpenTelemetry Spec configuration is incorrect, %s", err)
```

Comment:
```suggestion
			return fmt.Errorf("the OpenTelemetry Spec Prometheus configuration is incorrect, %s", err)
```

There are a lot of things that could be incorrect, we should give some hint as to where the problem lies.